### DISCUSSION TITLE
Disable red border for blocked large media elements

### DISCUSSION THREAD

**Discussion by vvkdev** [uBlockOrigin/uBlock-issues#3407] [Q&A] [answered]
When I use no-large-media option, all image placeholders have red dashed border.
How to change the color or style of this?
It seems that I used to add a string "*##img:style(border: 1px solid lightgrey !important;)" to My filters and it worked, but now I can't repeat it.
Thank you!

**Comment by krystian3w** [comment, marked as answer]
Due CSS specificity:

- https://developer.mozilla.org/en-US/docs/Web/CSS/Specificity
- https://specificity.keegan.st/

You should try this:

```adb
*##:root img:not(#style_important):style(border: 1px solid lightgrey !important;)
! Rest of blocked huge media (optional):
*##:root picture:not(#style_important):style(border: 1px solid lightgrey !important;)
*##:root audio:not(#style_important):style(border: 1px solid lightgrey !important;)
*##:root video:not(#style_important):style(border: 1px solid lightgrey !important;)
```

If too many photos will be bordered (e.g. smaller than limit or from cache loaded) then maybe needed use:
- `:matches-attr(/^data-.*/=/^$/)` to found `data-.*` with empty value (still not work for me with latest uBo DEV even garry idea), or
- `:matches-css(border: 2px dotted rgb(255, 0, 0))` to found stock border (still not work for me with latest uBo DEV), or copied by webmaster due inspiration of uBo.

due looks out of control in `uiStyles` now.

**Reply by garry-ut99** [reply]
> You should try this:

Works, but can be simplified to (or optionally `:where()` instead of `:is()`): 
```adb
*##:root :is(img,picture,audio,video):not(#style_important):style(border: 1px solid lightgrey !important;)
```

<hr>

> `:matches-attr(/^data-*/=/^$/)` to found `data-*` with empty value (not work for me with latest uBo DEV)

Doesn't work form me as well (btw  `-*` still matches, but only by luck, as it's wrong coz it works differently in regex):
```adb
*##:root :is(img,picture,audio,video):not(#style_important):matches-attr(/^data-[a-z0-9]{9}$/=""):style(border: 1px solid lightgrey !important;)
```

> `:matches-css(border: 2px dotted RGB(255, 0, 0))` to found stock border  (not work for me with latest uBo DEV)

Doesn't work form me as well (btw, your `matches-css` value doesn't match anything, but it still doesn't work even with a matching selector):
```adb
*##:root :is(img,picture,audio,video):not(#style_important):matches-css(border: /\dpx dotted rgb\(255, 0, 0\)/):style(border: 1px solid lightgrey !important;)
```

**Reply by krystian3w** [reply]
Corrected regexes are not work for me (I can only check with chromium in future), Seemingly I can see them in the logger and DOM inspector but that's not the point.

I guess:
- the sheet with procedural filtering is loaded lower than the styles from the add-on or raw CSS selectors,
-  needed is `:watch-attr` for `:matches-attr`, or
- thats attributes and CSS are now protected to dected itself with see change.

**Reply by garry-ut99** [reply]
> Corrected regexes are not work for me

Very unclear and confusing what do you mean by this, you should be more clear as for the context.
**a)** That regex don't work: don't match any DOM elements? 
**b)** Or that the filters with regex don't work: (no color change)?

In case **a)**: 
They do work on my side, they match elements on both Firefox & Chromium, example URL: `https://wiadomosci.onet.pl/opole/poszli-do-lasu-na-grzyby-zaskakujacy-final/ylk9mdg` screenshoots: [SCREENSHOOT 1](https://github.com/user-attachments/assets/ce4c88a9-b67b-4baa-ad09-b02ffdfde256), [SCREENSHOOT 2](https://github.com/user-attachments/assets/aed61d00-afce-4a04-b223-91f3a1a0fafc), provide your screenshoots as a proof they don't match anything on your side.

In case **b)**:  
1] You should have been wording more carefully, and you should have written "filters" or "filters with regex", and not "regex".
2] Nowhere I claimed my filters with regex work, on a contrary - examples I provided were examples of filters that **DON'T** work, see: "**Doesn't work** form me as well (...) : filter" and notice that the filter occurs after the colon, and the colon indicates what "doesn't work", you misread it as a filter that was supposed to work compared to your filter - but this was not an intention of my example filters.
